packages/material-renderers/test/renderers/MaterialArrayLayout.test.tsx

```diff
@@ -738,4 +910,236 @@ describe('Material array layout', () => {
+      </JsonFormsStateProvider>
+    );
+
+    expect(wrapper.find(MaterialArrayLayout).length).toBeTruthy();
+
+    expect(getChildLabel(wrapper, 0)).toBe('MSG_TYPE_1');
+  });
+
+  it('should render configured enum child label property as translated label', () => {
+    const core = initCore(
+      enumSchema,
+      uiSchemaWithEnumOrOneOfChildLabelProp,
+      enumOrOneOfData
+    );
+
+    // Enum Case - Translation
+    wrapper = mount(
+      <JsonFormsStateProvider
+        initState={{
+          renderers: materialRenderers,
+          core,
+          i18n: { translate: testTranslator },
+        }}
+      >
+        <MaterialArrayLayout
+          schema={enumSchema}
+          uischema={uiSchemaWithEnumOrOneOfChildLabelProp}
+        />
+      </JsonFormsStateProvider>
+    );
+
+    expect(wrapper.find(MaterialArrayLayout).length).toBeTruthy();
+
+    expect(getChildLabel(wrapper, 0)).toBe(
+      'translator.root.messageType.MSG_TYPE_1'
+    );
+  });
+
+  it('should render configured oneOf child label property as-is if no translator', () => {
+    const core = initCore(
+      oneOfSchema,
+      uiSchemaWithEnumOrOneOfChildLabelProp,
+      enumOrOneOfData
+    );
+
+    // OneOf - No translation
+    wrapper = mount(
+      <JsonFormsStateProvider
+        initState={{ renderers: materialRenderers, core }}
+      >
+        <MaterialArrayLayout
+          schema={oneOfSchema}
+          uischema={uiSchemaWithEnumOrOneOfChildLabelProp}
+        />
+      </JsonFormsStateProvider>
+    );
+
+    expect(wrapper.find(MaterialArrayLayout).length).toBeTruthy();
+
+    expect(getChildLabel(wrapper, 0)).toBe('MSG_TYPE_1');
+    expect(getChildLabel(wrapper, 1)).toBe('Second message type');
+  });
+
+  it('should render configured oneOf child label property as translated label', () => {
+    const core = initCore(
+      oneOfSchema,
+      uiSchemaWithEnumOrOneOfChildLabelProp,
+      enumOrOneOfData
+    );
+
+    // OneOf Case - Translation
+    wrapper = mount(
+      <JsonFormsStateProvider
+        initState={{
+          renderers: materialRenderers,
+          core,
+          i18n: { translate: testTranslator },
+        }}
+      >
+        <MaterialArrayLayout
+          schema={oneOfSchema}
+          uischema={uiSchemaWithEnumOrOneOfChildLabelProp}
+        />
+      </JsonFormsStateProvider>
+    );
+
+    expect(wrapper.find(MaterialArrayLayout).length).toBeTruthy();
+
+    expect(getChildLabel(wrapper, 0)).toBe(
+      'translator.root.messageType.MSG_TYPE_1'
+    );
+    // TODO Possible impact to existing jsonForm setup, previously if the translate fonction was setup,
+    //  the enumOf title was written as-is and not pushed to the translation function with a key.
+    //  The oneOf title is pushed for translation over the const value in renderer.ts/oneOfToEnumOptionMapper(),
+    //  seems a bug to me
+    expect(getChildLabel(wrapper, 1)).toBe(
+      'translator.root.messageType.Second message type'
+    );
```

Comment:
I came across this while fixing the code, I think it needs your opinion.

Reply:
The title is preferred over the value as in practice the value is often a non-string like an object while the title will always be a string. If that is not wanted, the user can define an `i18n` key in the schema.